### ISSUE TITLE
ERROR: Cannot migrate old registry: No remote 'bincrafters' defined in remotes

### ISSUE THREAD

**Issue by DoDoENT** [conan-io/conan#5099]
After updating conan from v1.14.5 to v1.15.0 and running `conan --version` to verify the installation, I get the following message:

```
$ conan --version
WARN: Migration: Updating settings.yml
WARN: ****************************************
WARN: settings.yml is locally modified, can't be updated
WARN: The new settings.yml has been stored in: /Users/dodo/.conan/settings.yml.new
WARN: ****************************************
WARN: registry.json has been deprecated. Migrating to remotes.json
ERROR: Cannot migrate old registry: No remote 'bincrafters' defined in remotes
```

This also happens on any subsequent `conan install` commands, which makes it unusable.

I don't have bincrafters remote - only conan-center and my local instance of Artifactory. Furthermore, my CI servers only use local instance of Artifactory (our policy is that all 3rd party packages first need to be approved by admin and imported to local Artifactory instance before they can be used in our codebase).

**Comment by memsharded** [comment]
This is weird. For some reason you have some trailing package in ``registry.json`` pointing to a remote that you no longer have. Could you please try manually removing from that file such references? I will have a look how to fix that migration, thanks!

**Comment by DoDoENT** [comment]
Thanks! Indeed there was a rouge entry in my `registry.json`. After manually removing it, the migration worked.

But nevertheless, I think the migration script should be robust to such cases.

**Comment by memsharded** [comment]
Closed by https://github.com/conan-io/conan/pull/5103, will be released in Conan 1.15.1. Thanks for reporting.
